src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,119 @@
 public class StringManipulation implements StringManipulationInterface {
+    private String myString;
 
     @Override
     public String getString() {
-        return null;
+        // Simple check if the String is null or not
+        if (myString == null){
+            return null;
+        }
+        else {
+            return myString;
+        }
     }
 
     @Override
     public void setString(String string) {
+        myString = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        //edge cases if String length equals 0
+        if (myString == null || myString.isEmpty()) {
+            return 0;
+        }
+        //excluding all the white space, then add each word into Array
+        String[] words= myString.split("\\s+");
+
+        //return the Array length, which is also the length of the string now
+        return words.length;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        // Position start at 1, not like index, so throws than error
+        if (n <= 0) {
+            throw new IllegalArgumentException("'n' must be greater than zero.");
+        }
+
+        // n must be smaller than the length of the string
+        if (n > myString.length()) {
+            throw new IndexOutOfBoundsException("'n' is greater than the string length.");
+        }
+
+        //initial an empty string
+        String result = "";
+        int position = 1;
+
+        for (int i = 0; i < myString.length(); i++) {
+            char current = myString.charAt(i);
+
+            //this is to find the 2,4,6 if n=2 or 3,6,9 if n=3,...
+            if (position % n != 0) {
+                result += current;
+            }
+
+            //if maintainSpacing is true then replace that character with a space
+            else if (maintainSpacing) {
+                result += ' ';
+            }
+
+            position++;
+        }
+
+        return result;
     }
 
     @Override
-    public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+    public String[] getSubStrings(int startWord, int endWord){
+        // check if the beginning word and end word less than 0 and conflict with each other
+        if (startWord <= 0 || endWord <= 0 || startWord > endWord) {
```

Comment:
I think these exceptions should be separated out. Currently, the exception throw doesn't differentiate between them. This means that for a user they won't know which part was wrong or how to fix it. Not a major issue, more a QoL improvement.